src/handlers/mod.rs

```diff
@@ -71,6 +72,17 @@ delegate_text_input_manager!(State);
 
 impl InputMethodHandler for State {
     fn new_popup(&mut self, surface: PopupSurface) {
+        if let Some((_, output)) = surface
+            .get_parent()
+            .and_then(|parent| self.niri.layout.find_window_and_output(&parent.surface))
+        {
+            let scale = output.current_scale().integer_scale();
+            let transform = output.current_transform();
+            let wl_surface = surface.wl_surface();
+            with_states(wl_surface, |data| {
+                send_surface_state(wl_surface, data, scale, transform);
```

Comment:
Why do IME popups get the state in new_popup, but Xdg popups get it on first commit, wouldn't it make sense to unify those? Also what happens when an IME popup changes its parent?

Reply:
ime popup changes its parent during` new_popup`, that's the reason the logic is here.

Reply:
xdg popup on the other side has a fixed parent which must be specified on `commit` and then it follows the parent window, so it gets events due to `with_surfaces` calls.

Reply:
I see, thanks.